tests/test_window_streams.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+import numpy as np
+from brainflow.board_shim import BoardShim, BrainFlowInputParams, BoardIds
+from neuros.main import WindowConfig, stream_windows, board_stream
+
+
+def test_window_config_validation():
+    """Test window configuration validation"""
+    # Valid configurations
+    WindowConfig(window_ms=500.0, overlap_ms=250.0)  # Known good values
+    WindowConfig(window_ms=100.0, overlap_ms=60.0)  # Another known good set
+
+    # Invalid configurations
+    with pytest.raises(ValueError):
+        WindowConfig(window_ms=100.0, overlap_ms=100.0)  # Equal window and overlap
+    with pytest.raises(ValueError):
+        WindowConfig(window_ms=100.0, overlap_ms=150.0)  # Overlap > window
+
+
+def test_basic_window_shape():
```

Comment:
Can you rephrase some of these test to make it more clear what is the scenario and expectation?
The Given...When...Then paradigm in the example at https://cucumber.io/ is a pretty solid pattern

Reply:
Will do, thanks for the cucumber.